### DISCUSSION TITLE
Having Trouble in Secton 3: Remix Funding - Trouble Moment = Deploying demo DataConsumerV3 Contract from Chainlink

### DISCUSSION THREAD

**Discussion by RenSylvain** [Cyfrin/foundry-full-course-cu#2817] [Q&A] [answered]
Hi,

After step 6 on this page: https://docs.chain.link/data-feeds/getting-started , I get this message:
(I better elaborate, step 6 is deploying a Data feed smart contract provided by Chainlink to show how data feeds work, the error is:) 

"Gas estimation failed
Gas estimation errored with the following message (see below). The transaction execution will likely fail. Do you want to force sending?

Failed to serialize transaction: toAddressIsNull"

and then it gives me an option to "send transaction" or "cancel transaction",

I've tried sending the transaction regardless of the warning, but an error in the console comes up:

"creation of DataConsumerV3 pending...
creation of DataConsumerV3 errored: Error occurred:  Internal JSON-RPC error..

 Internal JSON-RPC error.

If the transaction failed for not having enough gas, try increasing the gas limit gently."

I'll attach a screenshot of the environment I am working in for extra info.

Any suggestions for how I can successfuly deploy this Chainlink DataConsumerV3 contract without the error/roadblock?

<img width="1440" alt="Screen Shot 2024-11-05 at 11 09 55 AM" src="https://github.com/user-attachments/assets/99e059c5-8192-4ada-b591-20f1291d49dc">

**Comment by Kunnu01** [comment, marked as answer]
I think you are on the wrong network, `Custom (300) Network`, which is ZkSynk. You need to change it to Sepolia.

Once you change your network to Sepolia Testnet, it should work fine.

**Reply by RenSylvain** [reply]
Amazing. This was my first question on the discussions page. Thank you for such a quick answer.

I thought i had switched to Sapolia using InjectedWeb3, but I tried WalletConnect instead and that did the trick. I saw "Custom (300) Network" Switch to "Sepolia (11155111) network".

Appreaciate the help Kunnu01.

**Reply by Kunnu01** [reply]
You're welcome, King. Keep crushing 🚀

**Comment by EngrPips** [comment]
Good job here.
